### ISSUE TITLE
Avoid many info messages about whitespace

### ISSUE THREAD

**Issue by dkager** [snaekobbi/pipeline-mod-dedicon#29]
Could these be suppressed, either by tweaking the CSS or by flagging these as DEBUG?

```
[INFO]      white-space:pre-wrap could not be applied to target-counter(page)
```

And the CSS:

```
#generated-volume-toc li > a::after {
    /* The next line prevents whitespace from being collapsed. */
    white-space: pre-wrap;
    content: '  ' target-string(attr(href), print-page-to) '  ' target-counter(attr(href), page);
}
```

The reason for pre-wrap is to get the non-breaking spaces.

**Comment by bertfrees** [comment]
You could use real non-breaking space characters. I would think it should also be no problem to support `white-space:pre-wrap` on `target-counter()`, have to look into why it is currently not.

**Comment by dkager** [comment]
I don’t strictly need it inside the counter, though it would allow for some interesting CSS rules that could be very useful. E.g. to add the whitespace to the counter so that if the counter is missing you don’t get bogus whitespace.

**Comment by bertfrees** [comment]
No, but just to avoid the messages, because you only set the property on all of the segments or none.

**Comment by dkager** [comment]
Well that too, but if it actually works that’d be a nice bonus. ☺

**Comment by bertfrees** [comment]
Yes it should work, otherwise the message should stay :) I just don't think it will actually have any use, because a counter will rarely have any space in it. In the example you gave with the empty counter the `white-space` property does not have any use.

**Comment by dkager** [comment]
My use case was that if the chapter name is Foo, the braille page is 2, and the print counter is empty, you get:

```
'Foo' + '  ' + '' + '  ' + '2' = 'Foo    2'
```

To avoid that I tried appending the whitespace to the string-set in the pagenum rule, but that didn't work.

**Comment by bertfrees** [comment]
Ah I see what you mean, you could indeed append the whitespace to the string-set of the print page and in that case if white-space:pre-wrap would work on target-string() this should give you the desired result.

**Comment by bertfrees** [comment]
But you don't even need white-space if you just replace the regular spaces with no-break spaces (U+00A0) (also in the string-set).

**Comment by dkager** [comment]
Yup, that is what I did now.

**Comment by dkager** [comment]
Hyphens in target-counter(page) is also not supported, which makes sense. Tweaking CSS is probably the best 'fix'.
